manifests/init.pp

```diff
@@ -130,7 +114,7 @@
 class simp (
   Enum['simp','simp_lite','poss'] $scenario                   = 'simp',
   Boolean                         $enable_data_includes       = true,
-  String                          $data_includes_array_name   = 'classes',
+  Optional[Array]                 $classes                    = lookup('simp::classes', Optional[Array], { 'strategy' => 'deep', 'knockout_prefix' => '--' }, undef),
```

Comment:
Oh, the knockout prefix does work with arrays but you have to take care that people don't mess it up and leave things with '--' in them in the final result.